src/awscdk/cdk-config.ts

```diff
@@ -107,15 +119,37 @@ export class CdkConfig extends Component {
         output: this.cdkout,
         build: options.buildCommand,
         watch: {
-          include: options.watchIncludes ?? [],
-          exclude: options.watchExcludes ?? [],
+          include: () => this.include,
+          exclude: () => this.exclude,
         },
       },
     });
 
     project.gitignore.exclude(`/${this.cdkout}/`);
     project.gitignore.exclude(".cdk.staging/");
   }
+
+  /**
+   * Add includes to `cdk.json`.
+   * @param patterns The includes to add.
+   */
+  public addIncludes(patterns: string | string[]) {
+    if (!Array.isArray(patterns)) {
+      patterns = [patterns];
+    }
+    this.include.push(...patterns);
+  }
```

Comment:
Let's make this `string[]` instead:
```suggestion
  public addIncludes(...patterns: string[]) {
    this.include.push(...patterns);
  }
```

We use a tool named jsii to translate projen's API into other languages like Java - and since they cannot express type unions as easily, other users will just see a method that accepts "Object" instead of "string" or "Array\<string>".

Reply:
Ah interesting, is there a way to make jsii throw an error or warning when union types are used?

Reply:
That's a good question - I'm not sure off the top of my head, but I'll take a look into it :)